go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -334,9 +334,9 @@ func (qre *QueryExecutor) Stream(callback StreamCallback) error {
 			return err
 		}
 		defer txConn.Unlock()
-		if len(qre.settings) > 0 {
-			if err = txConn.ApplySettings(qre.ctx, qre.settings); err != nil {
-				return vterrors.Wrap(err, "failed to execute system settings on the connection")
+		if qre.setting != nil {
```

Comment:
Do we need a `IsNil()` test too here?

Reply:
it is only needed when you convert a struct reference to interface. So, it is not applicable here.